packages/framework-provider-azure-infrastructure/src/infrastructure/synth/application-synth.ts

```diff
@@ -151,34 +154,38 @@ export class ApplicationSynth {
       'graphql'
     )
 
-    const functionAppDataResource = TerraformWebPubSubExtensionKey.build(
-      this.config,
-      azurermProvider,
-      this.terraformStackResource,
-      resourceGroupResource,
-      functionAppResource,
-      this.appPrefix
-    )
-
-    const webPubSubHubResource = TerraformWebPubsubHub.build(
-      azurermProvider,
-      this.terraformStackResource,
-      resourceGroupResource,
-      webPubSubResource,
-      this.appPrefix,
-      functionAppResource,
-      functionAppDataResource,
-      hubName
-    )
+    let webPubSubHubResource
+    if (this.config.enableSubscriptions) {
+      const functionAppDataResource = TerraformWebPubSubExtensionKey.build(
+        this.config,
+        azurermProvider,
+        this.terraformStackResource,
+        resourceGroupResource,
+        functionAppResource,
+        this.appPrefix
+      )
 
+      if (webPubSubResource) {
+        webPubSubHubResource = TerraformWebPubsubHub.build(
+          azurermProvider,
+          this.terraformStackResource,
+          resourceGroupResource,
+          webPubSubResource,
+          this.appPrefix,
+          functionAppResource,
+          functionAppDataResource,
+          hubName
+        )
+      }
```

Comment:
Same here. So the question is: Could `this.config.enableSubscriptions` be true, but `webPubSubResource` be undefined? I guess it could not, so I would write the code in a way that it is impossible to happen. What I mean is to just ask for `webPubSubResource` in the main if, and not for `this.config.enableSubscriptions`. 

My preference would be to ask always for the `this.config.enableSubscriptions`, but as the wePubSubResource can be undefined, then we would get a compilation error

Reply:
Or! Maybe this is a better idea for this comment and the previous one: Always ask for `this.config.enableSubscriptions`, and then throw an error if the required resources are undefined, like `webPubSubResource` in this case. 

This is closer to the truth as if we are a the situation where `this.config.enableSubscriptions` is true but `webPubSubResource` is undefined, then there was some error, and it is better to show it to the user instead of ignoring it